### DISCUSSION TITLE
Go to declaration does not work

### DISCUSSION THREAD

**Discussion by speron26** [pulsar-edit/.github#291] [Core Application & Packages]
Hi all,

Is it possible to "go to declaration" of a function or an object in Pulsar-Edit ... 

I tried many keybindings but nothing works ... (  eg ctrl+shift+click )

Is it possible to have such a function in the core application ? or do i need to install a specific package ?

Thanks a lot for your help

Best Regards
Stéphane

**Comment by savetheclocktower** [comment]
It depends on your language. In general you need a “brain” that knows where a function is defined.

The “Go to Declaration” command is in the built-in `symbols-view` package and the binding defaults to <kbd>Cmd+Opt+Down</kbd> on macOS and a similar shortcut on other platforms.

But  you'll want to install an IDE package to enjoy this feature. For instance, TypeScript and JavaScript files can gain jump-to-declaration functionality once [pulsar-ide-typescript-alpha](https://web.pulsar-edit.dev/packages/pulsar-ide-typescript-alpha) is installed. If you're writing C/C++, [pulsar-ide-clangd](https://web.pulsar-edit.dev/packages/pulsar-ide-clangd) does what you need. For Python, you want [pulsar-ide-python](https://web.pulsar-edit.dev/packages/pulsar-ide-python). And so on.

Once you have an IDE package to tell Pulsar where an arbitrary function is defined, you should also be able to hyperclick on any symbol in the document and be taken to its declaration; this requires a package like [hyperclick](https://web.pulsar-edit.dev/packages/hyperclick).
